azurerm/internal/services/appplatform/spring_cloud_config_server.go

```diff
@@ -0,0 +1,129 @@
+package appplatform
+
+import (
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/validation"
+)
+
+func SchemaConfigServerHttpBasicAuth(conflictsWith ...string) *schema.Schema {
+	s := &schema.Schema{
+		Type:     schema.TypeList,
+		Optional: true,
+		MaxItems: 1,
+		Elem: &schema.Resource{
+			Schema: map[string]*schema.Schema{
+				"username": {
+					Type:         schema.TypeString,
+					Required:     true,
+					ValidateFunc: validation.StringIsNotEmpty,
+				},
+
+				"password": {
+					Type:      schema.TypeString,
+					Required:  true,
+					Sensitive: true,
+				},
+			},
+		},
+	}
+	if len(conflictsWith) > 0 {
+		s.ConflictsWith = conflictsWith
+	}
+	return s
+}
+
+func SchemaConfigServerSSHAuth(conflictsWith ...string) *schema.Schema {
+	s := &schema.Schema{
+		Type:     schema.TypeList,
+		Optional: true,
+		MaxItems: 1,
+		Elem: &schema.Resource{
+			Schema: map[string]*schema.Schema{
+				"private_key": {
+					Type:         schema.TypeString,
+					Required:     true,
+					Sensitive:    true,
+					ValidateFunc: validation.StringIsNotEmpty,
+				},
+
+				"host_key": {
+					Type:         schema.TypeString,
+					Optional:     true,
+					Sensitive:    true,
+					ValidateFunc: validation.StringIsNotEmpty,
+				},
+
+				"host_key_algorithm": {
+					Type:     schema.TypeString,
+					Optional: true,
+					ValidateFunc: validation.StringInSlice([]string{
+						"ssh-dss",
+						"ssh-rsa",
+						"ecdsa-sha2-nistp256",
+						"ecdsa-sha2-nistp384",
+						"ecdsa-sha2-nistp521",
+					}, false),
+				},
+
+				"strict_host_key_checking_enabled": {
+					Type:     schema.TypeBool,
+					Optional: true,
+				},
```

Comment:
The schema will default this to `false`, the docs say the Service default is `true`.
```suggestion
				"strict_host_key_checking_enabled": {
					Type:     schema.TypeBool,
					Optional: true,
					Default:  true,
				},
```
I don't believe this will need any other changes to support.

Reply:
the docs says it default to true, but in the portal it seems the default value is false. Of course either is ok. By the way, should we add `Defaults to true` in the resource doc?